Shape_detection/include/CGAL/Shape_detection/Region_growing/Polygon_mesh/Least_squares_plane_fit_region.h

```diff
@@ -383,25 +415,6 @@ namespace Polygon_mesh {
       return Point_3(x, y, z);
     }
 
-    // Compute normal of the face.
-    template<typename Face>
-    Vector_3 get_face_normal(const Face& face) const {
```

Comment:
I think the issue encountered by @lrineau would be fixed by only fixing this function. We should look for 3 non-collinear vertices to get a normal. I would then skip the proper 2D CDT and keep the centroid triangulation for the plane fitting (only matters for non exactly coplanar points anyway). @afabri what do you think?

Reply:
I also thought that for computing a face normal it is enough to find 3 sufficiently non-collinear points, but Sven told  me that for non-convex faces he has to compute some weight and for that he needs a triangulation.  We have to check with him.

Reply:
These were two separate problems:
1. get_face_normal is used for fitting a plane to a single face and for checking whether a face fits to an adjacent region. This was likely the issue encountered by @lrineau.
2. PCA was not using triangulated faces.